core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -953,6 +953,9 @@ private Object readResolve() {
     @Nonnull
     @Restricted(NoExternalUse.class)
     public InstallState getInstallState() {
+        if (installState == null || installState.name() == null) {
+            return InstallState.UNKNOWN;
```

Comment:
@kzantow But it is a special case, right? This cases are candidates to be mentioned in the log system. At least, this is my opinion.